README.md

```diff
@@ -217,29 +217,70 @@ Authors should respect the `prefers-reduced-motion` media query by conditionally
-- Describe reversing transitions on back with bfcache and adopted portals.
-- Outline our solution for these, in terms of observable effects for users (not in terms of spec primitives).
-- Describe an option for activating portals with replacement.
-- Describe an option for activating adopted portals which traverses session history.
+If the predecessor is not adopted, then it may enter back/forward cache (bfcache). If the predecessor can't be cached then it is unloaded.
+
+In addition to enabling seamless transitions to portaled content, we also want to enable seamless transitions from previously portaled content, the successor, back to its predecessor. This is done by re-hosting the successor context in its original portal element, and can be thought of as a form of implicit adoption. When a live predecessor is navigated to (e.g. by the user pressing the back button), we restore the successor context to its portal element in the predecessor. We then fire a `restoration` event on the element so that the predecessor can respond (e.g. by reversing the animation before the original activation). The live predecessor may be taken from bfcache, or if the predecessor was adopted, taken from a portal in the successor. In the latter case, this involves an implicit activation of the portal in the successor. Also note that since this restoration involves turning a top-level browsing context back into a portal context, only limited forms of restoration are proposed for the cross-origin case, as described above for regular adoption.
+
+If the predecessor page can't handle being restored, it may set the `irreversible` field in the activation options as an escape hatch to ensure it's unloaded. This prevents adoption and causes the predecessor to be ineligible for bfcache. Furthermore, eligibility for bfcache is a requirement for being adoptable.
+
+Since whether a portal element is hosting a context changes over time, the element itself can be thought of as having a lifecycle. This is exposed to the page via `state`.
+
+State | Meaning
+----- | -------
+empty | Nothing has ever been loaded in the portal, or the previous contents were discarded due to failed restoration, the user agent reclaiming resources, etc.
+live | The portal is hosting a browsing context.
+activated | The portal has activated. This host page is adopted or in bfcache.
+frozen | The portal is hosting a browsing context, but that context has been [frozen](https://github.com/WICG/page-lifecycle).
+epitaph | The previous contents were discarded, but unlike the empty state, there is some visual representation of the previous content.
+
+The events fired on a portal element to notify of its lifecycle changes are as follows.
+
+Event | Meaning
+----- | -------
+onactivation | The portal contents have been activated by something other than an explicit call to the `activate` method (e.g. default click, back navigation).
+onrestoration | The previous contents were restored into this portal element.
```

Comment:
I think the typical event naming convention would be `onactivate` and `onrestore`

Reply:
Done.